pkg/controller/controller_utils.go

```diff
@@ -942,16 +942,6 @@ func PatchNodeTaints(c clientset.Interface, nodeName string, oldNode *v1.Node, n
 		return fmt.Errorf("failed to marshal old node %#v for node %q: %v", oldNode, nodeName, err)
 	}
 
-	newTaints := newNode.Spec.Taints
```

Comment:
Sorry, I think I wrote something that doesn't make sense.

Anyway - the change that you suggested doesn't seem correct. Because the only change we would like to do is to get the "tainst" field from the newNode and apply it to oldNode. 
With your change, if anything else changed in newNode, this will also be applied, which isn't something we want.